src/session/session_api.c

```diff
@@ -1703,7 +1724,17 @@ __wt_session_range_truncate(
     if (!F_ISSET(start, WT_CURSTD_KEY_INT)) {
         needs_next_prev = true;
         if (orig_start_key != NULL) {
-            WT_ERR_NOTFOUND_OK(start->search_near(start, &cmp), true);
+            ret = start->search_near(start, &cmp);
+            if (ret == WT_PREPARE_CONFLICT) {
+                /* Don't return a prepare conflict if the key we land on falls outside the truncate
+                 * range. */
+                if (__check_key_outside_truncate_range(start, trunc_info)) {
+                    F_SET(session->txn, WT_TXN_IGNORE_PREPARE);
```

Comment:
@ajmorton Can you clarify what you mean by setting `start` to 25? Do you mean the cursor in WT gets positioned to 25 while searching for some other key, and if so, which other key? Or do you mean the user it calling truncate with a start key of 25?

Reply:
Yep, sorry. As is the user wants to truncate from range 25-50 and therefore `truncate_info->start` wants to position on key 25, but because it's not present `search_near` finds the next best key, which in this case is either 20 `(prepared)` or `30`. 

I'd want to confirm that when `search_near` starts looking at smaller keys and encounters `20 (prepared)` that the `WT_TXN_IGNORE_PREPARE` flag means we turn around and start looking at larger keys - finding `30` - instead of skipping `20` and positioning on `10`

Reply:
We won't go back to 10, because the start cursor only moves forwards.

But that concern about accidentally removing 20 is definitely valid. I'll probably have to do some more code reading and/or test this.